### PR TITLE
[FBZ-10442] - Migrate Custom Cron Recipe to v7

### PR DESCRIPTION
**Pull request by kvbautista**
**Description of your patch**
Migrate Custom Cron Recipe to v7

**Recommended Release Notes**
Added support for custom-cron recipes

**Estimated risk**
Medium

**Components involved**
* `cookbooks/ey-cron/recipes/custom.rb`
* `custom-cookbooks/cron`

**Dependencies**
* `ey-lib`
* `ey-ntp`

**Description of testing done**
* Provision v7 stack environment
* Create custom cron recipe based on `custom-cookbooks/cron/cookbooks/custom-cron`
* Modify `ey-custom` to include `custom-cron`
* Upload to v7 environment
* Trigger an 'Apply' across the environment to execute chef run
* SSH into instance where cronjob is expected to exist
* Execute command `crontab -l` as the user

**QA Instructions**
* Provision v7 stack environment
* Create custom cron recipe based on `custom-cookbooks/cron/cookbooks/custom-cron`
* Modify `ey-custom` to include `custom-cron`
* Upload to v7 environment
* Trigger an 'Apply' across the environment to execute chef run
* SSH into instance where cron job is expected to exist
* Execute command `crontab -l` as the user
* Verify if cron job exists

### CODE REVIEW

**Comment by dvalfre** [comment]
@kdotbautista the relevant ticket is fbz10442 😉 - let's make sure the PR's title includes it.

**Comment by kvbautista** [comment]
[FBZ-10452] Added as commit `591e606`

File: cookbooks/ey-cron/recipes/custom.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  deleted_crons.each do |deleted_cron|
+    cron "custom_cron_#{deleted_cron}" do
+      user user
+      action :delete
+    end
+  end
+end
+
+if crontab_instance?(node)
+  if node["custom_crons"]; then
```
Comment (mushyy) [inline cookbooks/ey-cron/recipes/custom.rb]: Do not use then for multi-line if. (https://rubystyle.guide#no-then) 

Could this be correct to inline with cookstyle please or add an exception

**Review by mushyy — CHANGES_REQUESTED** [review]
Changes in conversation comments

File: cookbooks/ey-cron/recipes/custom.rb (hunk 1 above)
Comment (kvbautista) [reply cookbooks/ey-cron/recipes/custom.rb]: Fixed to be inline with cookstyle

From:

```
  if node["custom_crons"]; then
```

To:

```
  if node["custom_crons"]
```

**Comment by kvbautista** [comment]
Fixed `cookbooks/ey-cron/recipes/custom.rb`to be inline with cookstyle as mentioned by Callum

From:

```
  if node["custom_crons"]; then
```

To:

```
  if node["custom_crons"]
```

**Review by mushyy — APPROVED** [review]
Fantastic looks good to me

**Pull request merged**